### PR TITLE
Mission fixes and modder detection improvements

### LINKED ISSUE #1556: [Bug]: In First round of funding: garbage truck，Can‘t drive vehicle and rubbish truck
### Describe the bug

Unable to drive your own vehicle and the operator's garbage truck during the Fundraising Round: Garbage Truck segment

### Steps To Reproduce

Unable to drive your own vehicle and the operator's garbage truck during the Fundraising Round: Garbage Truck segment

### Expected Behavior

Unable to drive your own vehicle and the operator's garbage truck during the Fundraising Round: Garbage Truck segment

### Screenshots and/or Logs

```
<paste your logs here>
```

### Store

Steam

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### LINKED ISSUE #1559: [Bug]: The robbery mission cannot drive the specified vehicle（prison Break and First round of fundraising）
### Describe the bug

The robbery mission cannot drive the specified vehicle（prison Break and First round of fundraising）

### Steps To Reproduce

The robbery mission cannot drive the specified vehicle（prison Break and First round of fundraising）

### Expected Behavior

The robbery mission cannot drive the specified vehicle（prison Break and First round of fundraising）

### Screenshots and/or Logs

```
<paste your logs here>
```

### Store

Steam

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### LINKED ISSUE #1569: [Bug]: After the task is over, re-entering will be stuck on the "initializing" of the apartment
### Describe the bug

After the task is over, re-entering will be stuck on the "initializing" of the apartment

### Steps To Reproduce

After finishing the robbery mission, it displayed "failed to join the battle" and automatically switched the battle situation, and then when I re-entered the robbery mission through the mobile phone, he was always stuck on the interface of the apartment and displayed "initialization"

### Expected Behavior

It should be to enter the battle after the end, use the mobile phone to jump directly to the task instead of "initialization"

### Screenshots and/or Logs

```
<paste your logs here>
```

### Store

Steam

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### PR DESCRIPTION
**Pull request by maybegreat48**
Fixes #1569 
Fixes #1559 
Fixes #1556 
Added new modder detections
Transition state UI now shows host broadcast received percent
Improved crash protections

### CODE REVIEW

File: src/backend/context/chat_command_context.cpp [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #include "chat_command_context.hpp"
 
+#include "fiber_pool.hpp"
```
Comment (Yimura) [inline src/backend/context/chat_command_context.cpp:3]: Random include added

File: src/hooks/protections/received_array_update.cpp [hunk 2]
```diff
@@ -4,6 +4,8 @@
 #include "gta_util.hpp"
 #include "hooking.hpp"
 #include "script_local.hpp"
+#include "services/players/player_service.hpp"
+#include "util/misc.hpp"
```
Comment (Yimura) [inline src/hooks/protections/received_array_update.cpp:8]: Unused includes?

**Review by Yimura — CHANGES_REQUESTED** [review]

File: src/backend/context/chat_command_context.cpp (hunk 1 above)
Comment (maybegreat48) [reply src/backend/context/chat_command_context.cpp:3]: I've removed an include from a header file and stuff that implicitly relied on it being included broke

File: src/hooks/protections/received_array_update.cpp (hunk 2 above)
Comment (maybegreat48) [reply src/hooks/protections/received_array_update.cpp:8]: I've removed an include from a header file and stuff that implicitly relied on it being included broke

Comment (Yimura) [reply src/hooks/protections/received_array_update.cpp:8]: Alright.

File: src/backend/context/chat_command_context.cpp (hunk 1 above)
Comment (Yimura) [reply src/backend/context/chat_command_context.cpp:3]: Alright.

**Review by Yimura — APPROVED** [review]

**Pull request merged**